### ISSUE TITLE
[FR] Conversion from CSV file to database

### ISSUE THREAD

**Issue by kazdonkai** [RafaelGB/obsidian-db-folder#129]
Hi,
Currently, we can convert from a database to a CSV file, but would you consider making it possible to do the reverse?
In that case, we would like the file to be automatically linked when the file name is specified.

**Comment by RafaelGB** [comment]
hi @kazdonkai 

Could be another source option and select a CSV inside your vault

**Comment by RafaelGB** [comment]
The problem with that is to updated the csv, so in fact, the import of csv file implies create a note per row

**Comment by kazdonkai** [comment]
I use an app that allows me to scan barcodes and register books.
That app can export bibliographic information to a CSV file, so I want to use this plugin to create a batch of notes with bibliographic information.

**Comment by RafaelGB** [comment]
added to the roadmap, sure :)

**Comment by ooker777** [comment]
have you tried this? https://github.com/kometenstaub/csv-to-md/

**Comment by RafaelGB** [comment]
hi @ooker777  I do not start with this one.

Your link is a Python library, I am afraid that just typescript is available for plugins (maybe in some years with pyscript :P)
https://pyscript.net/

**Comment by ooker777** [comment]
hmm no, I mean using that script to convert cvs to md first, then you can use the plugin to display data

Also, I also find this plugin: https://github.com/SkepticMystic/import-into-obsidian. Does it suit your need?

**Comment by RafaelGB** [comment]
I can not use python inside obsidian, thats the problem, Obsidian just allow a unique JS file with all the logic.

The plugin that you share could be an option, sure!

**Comment by lbgws2** [comment]
As for batch data import, I check whether this function is implemented every day

**Comment by RafaelGB** [comment]
hi @lbgws2, currently we are working on a big refactor of the plugin to a new version of react-table | react18 and will take some time (2months aprox)

The plugin grew faster than was expected and the rebuilding is a must.

Once the changes are implemented, this kind of features could be build easily and better

**Comment by lbgws2** [comment]
It's been a few months, and the data import I want has not been realized. Continue to pay attention

**Comment by RafaelGB** [comment]
Thanks for your patience @lbgws2 , the volume of issues is big and this kind of improvements requires its time. We need a stable version of the core features first

To realice the issue it's necessary an entire new library to map all the fields and support all the possible types. Think about a format (every row as a new file, control duplicates, a tuple of type,values etc...)

So its not in the roadmap in short terms right now. I want to prioritise the formulas

**Comment by RafaelGB** [comment]
first version


https://user-images.githubusercontent.com/11924043/189477663-945ab163-934b-4352-a489-abd8d76e7b2a.mov

**Comment by RafaelGB** [comment]
The import adds all the columns to the file and just show the defined columns of your table without it requires a refresh

**Comment by lbgws2** [comment]
Now, this function has come into reality from my dream. Thank you very much

**Comment by bennsche** [comment]
I'm happy that there is a import-function now, but i have some problems. For example I am not able to import an exported database.

I created a Row and the md file looks like this:
```
---
test_1: test 1
test_2: test 2
test_3: test 3
test_4: test 4
---
```

I exported the db to csv and this is the result:

```
"File","test 1","test 2","test 3","test 4"
"Verwaltung/Partituren/1.md","test 1","test 2","test 3","test 4"
```

After the import, the md file looks like this:

```
---
File: test 1
test 1: test 2
test 2: test 3
test 3: test 4
---
```

I tried to import some other clear csv-files, but i got the same issues. After the import it looks like a random key:value relation.
